packages/workflow-transformer-async/flows/ReactExample.js

```diff
@@ -0,0 +1,20 @@
+/* eslint-env node */
+import React from 'react';
+import render, { Workspace, requireComponent, createComponentRecursive } from 'workflow-react';
+import { Terminal } from 'workflow-apps-defaults';
+
+const TerminalAsync = createComponentRecursive({
+  type: 'async',
+  name: 'TerminalAsync',
+  loader: async () => Terminal,
```

Comment:
This resolves to `Terminal` in the transformer which runs after react rendering.

Reply:
Ah, that's true :/ 

Anyways this is a great first stab at the `async` functionality! :D And I'd rather sacrifice the consistency in `workflow-react` than in the `workflow`/`workflow-core`. 

I'm just thinking this might be a bit magical to a react user, is a leaky abstraction and might limit our flexibility in the React layer. But we'll probably figure it out or maybe throwing an exception in `render` might help us someday.